resources/assets/lib/beatmap-discussions/editor.tsx

```diff
@@ -179,6 +180,11 @@ export default class Editor extends React.Component<Props, State> {
     if (!this.props.editMode) {
       const content = JSON.stringify(value);
 
+      // prevent document from becoming empty (and invalid) - ideally this would be handled in `withNormalization`, but that isn't run on every change
+      if (value.length === 0) {
+        value = this.emptyDocTemplate;
+      }
```

Comment:
this should be done regardless of `editMode` prop as same problem happens when editing reviews